internal/testutil/storage_cleaner.go

```diff
@@ -28,15 +28,43 @@ import (
 	"google.golang.org/api/iterator"
 )
 
-// CreateTestBucket creates a new bucket with the given prefix
-func CreateTestBucket(ctx context.Context, t *testing.T, client *storage.Client, projectID, prefix string) (string, error) {
+// TestBucket creates a new bucket with the given prefix and registers a cleanup
+// function to delete the bucket and any objects it contains when the test finishes.
+// TestBucket returns the bucket name. It fails the test if bucket creation fails.
+func TestBucket(ctx context.Context, t *testing.T, projectID, prefix string) string {
+	t.Helper()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		t.Fatalf("storage.NewClient: %v", err)
+	}
+	t.Cleanup(func() { client.Close() })
+	return CreateTestBucket(ctx, t, client, projectID, prefix)
+}
+
+// CreateTestBucket creates a new bucket with the given prefix and registers a
+// cleanup function to delete the bucket and any objects it contains.
+// It is equivalent to TestBucket but allows Storage Client re-use.
+func CreateTestBucket(ctx context.Context, t *testing.T, client *storage.Client, projectID, prefix string) string {
```

Comment:
Maybe re-name these to indicate what the difference is (e.g. CreateBucket vs CreateBucketWithClient)? Also is it actually important to have both functions available? Creating an extra client in a test is not much overhead.

Reply:
So I don't want to change the signature of the `CreateTestBucket` function since it will break tests. But I agree that we could just create the extra client instead.